minio/select.py

```diff
@@ -420,13 +420,10 @@ def stream(self, num_bytes=32*1024):
         Stream extracted payload from response data. Upon completion, caller
         should call self.close() to release network resources.
         """
-        while True:
-            if self._payload:
+        while self._read() > 0:
```

Comment:
This problem never occurs because of https://github.com/minio/minio-py/blob/master/minio/select.py#L273. You will get an exception. If `RawIOBase.read()` returns less bytes than requested, but still not received EOF, you would need to fix `_read()` function.

Reply:
The `_read()` function works correctly, as far as I can tell. The bug is in the `stream` method. If you run the adjusted test (with `num_bytes=16`, which is less than the payload size), you'll see that it fails without the fix.